les/costtracker.go

```diff
@@ -94,21 +106,29 @@ type costTracker struct {
 
 	inSizeFactor, outSizeFactor float64
 	gf, utilTarget              float64
+	minBufLimit                 uint64
 
 	gfUpdateCh      chan gfUpdate
 	gfLock          sync.RWMutex
 	totalRechargeCh chan uint64
 
-	stats map[uint64][]uint64
+	stats                                        map[uint64][]uint64
+	logger                                       *csvlogger.Logger
+	logRecentUsage, logTotalRecharge, logRelCost *csvlogger.Channel
 }
 
-// newCostTracker creates a cost tracker and loads the cost factor statistics from the database
-func newCostTracker(db ethdb.Database, config *eth.Config) *costTracker {
+// newCostTracker creates a cost tracker and loads the cost factor statistics from the database.
+// It also returns the minimum capacity that can be assigned to any peer.
+func newCostTracker(db ethdb.Database, config *eth.Config, logger *csvlogger.Logger) (*costTracker, uint64) {
 	utilTarget := float64(config.LightServ) * flowcontrol.FixedPointMultiplier / 100
 	ct := &costTracker{
-		db:         db,
-		stopCh:     make(chan chan struct{}),
-		utilTarget: utilTarget,
+		db:               db,
+		stopCh:           make(chan chan struct{}),
+		utilTarget:       utilTarget,
+		logger:           logger,
+		logRelCost:       logger.NewMinMaxChannel("relativeCost", true),
+		logRecentUsage:   logger.NewMinMaxChannel("recentUsage", true),
+		logTotalRecharge: logger.NewChannel("totalRecharge", 0.01),
 	}
 	if config.LightBandwidthIn > 0 {
 		ct.inSizeFactor = utilTarget / float64(config.LightBandwidthIn)
```

Comment:
The `inSizeFactor` and `outSizeFactor` seems incorrect here. The flag description is `Incoming bandwidth limit for light server (1000 bytes/sec, 0 = unlimited)`. So if the `lightbwin` is 1000, it means if the total incoming traffic more than 1000bytes/sec, then the total buffer value will be drained.

I think it should be `ct.inSizeFactor = utilTarget * 1000 / float64(config.LightBandwidthIn)`

Reply:
So maybe the parameter description is misleading. What I meant is that LightBandwidthOut=1 means 1000bytes/sec limitation (that is the base unit for these parameters, not the default value). This is how the formula works right now.

Reply:
All right. The parameter description is really misleading.

Reply:
I changed it to "kilobytes/second", I hope it is clear now.